src/storage.rs

```diff
@@ -1233,21 +1234,33 @@ fn is_known_criteria(valid_criteria: &[CriteriaName], criteria_name: &CriteriaNa
 /// which could not be interpreted, due to issues such as being created with a
 /// newer version of cargo-vet.
 pub(crate) fn foreign_audit_file_to_local(
+    name: &str,
     foreign_audit_file: ForeignAuditsFile,
 ) -> ForeignAuditFileToLocalResult {
     let mut ignored_criteria = Vec::new();
-    let mut criteria: SortedMap<CriteriaName, CriteriaEntry> = foreign_audit_file
-        .criteria
-        .into_iter()
-        .filter_map(|(criteria, value)| match parse_imported_criteria(value) {
-            Some(entry) => Some((criteria, entry)),
-            None => {
-                ignored_criteria.push(criteria);
-                None
-            }
-        })
-        .collect();
-    let valid_criteria: Vec<CriteriaName> = criteria.keys().cloned().collect();
+    let valid_criteria: Vec<CriteriaName>;
+    let mut criteria: SortedMap<CriteriaName, CriteriaEntry>;
+    if foreign_audit_file.criteria.is_empty() {
+        warn!("No criteria definitions found when importing from `{name}`");
+        criteria = SortedMap::new();
+        valid_criteria = vec![];
+    } else {
+        criteria = foreign_audit_file
+            .criteria
+            .into_iter()
+            .filter_map(|(criteria, value)| match parse_imported_criteria(value) {
+                Some(entry) => Some((criteria, entry)),
+                None => {
+                    ignored_criteria.push(criteria);
+                    None
+                }
+            })
+            .collect();
+        valid_criteria = criteria.keys().cloned().collect();
+        if valid_criteria.is_empty() {
+            warn!("All criteria definitions failed to parse when importing from `{name}`");
+        }
```

Comment:
The intention of `ForeignAuditFileToLocalREsult` is that it contains the relevant information (which criteria were ignored) in the returned data structure.

I would prefer to not add this extra logging in this code. If we want to add some form of error logging about this, I think it would be preferable to e.g. record how many criteria were present upstream in the `ForeignAuditFileToLocalResult` so the caller could log something like `{x} of {y} foreign custom criteria failed to parse when importing from {name}: {list}`.

If we want to add a `warn!` trace log within this function, I think I'd want it to be don in that style and based on `ignored_criteria` - perhaps something like:

```rust
if !ignored_criteria.is_empty() {
  warn!("{} of {} custom criteria failed to parse when importing from `{}`",
        ignored_criteria.len(), foreign_audit_file.criteria.len(), name);
}
```

Reply:
This is already done (almost, it doesn't include `audit_file.criteria.len()` as in your proposal, but it seems ok):

```
if !ignored_criteria.is_empty() {
    warn!(
        "Ignored {} invalid criteria entries when importing from '{}'\n\
        These criteria may have been made with a more recent version of cargo-vet",
        ignored_criteria.len(),
        name
    );
    info!(
        "The following criteria were ignored when importing from '{}': {:?}",
        name, ignored_criteria
    );
}
```

So, I went ahead and reverted the extra warnings from the beginning of `fn foreign_audit_file_to_local`.  The new warnings were based on a misunderstanding on my side - I've incorrectly thought that there needs to be at least one valid custom criteria definition (because I didn't think about `safe-to-deploy` and/or `safe-to-run`).